sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -594,6 +620,8 @@ kernel_pid_t gcoap_init(void)
 
 void gcoap_register_listener(gcoap_listener_t *listener)
 {
+    _check_listener(listener);
```

Comment:
if (as suggested above) check_listener would have a return value, you could use `assert(!_check_listener(listener);` here.